src/java.base/share/classes/java/security/DomainLoadStoreParameter.java

```diff
@@ -134,8 +134,7 @@ public DomainLoadStoreParameter(URI configuration,
             throw new NullPointerException("invalid null input");
         }
         this.configuration = configuration;
-        this.protectionParams =
-            Collections.unmodifiableMap(new HashMap<>(protectionParams));
+        this.protectionParams = Map.copyOf(protectionParams);
```

Comment:
This change is no longer strictly compliant with the specification. On line 126, it says "It is cloned to prevent subsequent modification." Now, if the set passed in is unmodifiable, that is no longer true. It also will now throw NPE if any of the keys or values are null, whereas before it did not, and the specification does not specify that in the @throws clause.

So, this change would require specification changes and a CSR.

Can you provide more details on whether this is a real performance issue in practice? 

I would tend to avoid changes like this that affect the API specification unless there is a very strong compelling case where it improves performance for a real use case.

Reply:
> it says "It is cloned to prevent subsequent modification."
> Now, if the set passed in is unmodifiable, that is no longer true. 

My understanding:
The parameter `protectionParams` is cloned as `this.protectionParams`,
and `this.protectionParams` should be prevented from subsequent modification.

`Map::copyOf` looks meet this requirement, no matter the parameter `protectionParams` is modifiable or not.

> It also will now throw NPE if any of the keys or values are null, whereas before it did not,
> and the specification does not specify that in the @throws clause.

Yes, this is a problem.
I don't want to change any behavior.

Reply:
> The latter part is true (prevented from subsequent modification) but, unless I am mistaken, the former (making a clone/copy) is not. For example, before your change, this assert would pass:
> 
> ```
> Map m = Collections.unmodifiableMap(map);
> DomainLoadStoreParameters params = new DomainLoadStoreParameters(uri, m);
> assert m != params.getProtectionParams();
> ```
> 
> After your change, I think it fails (can you check?). Even though the protection in both cases should be adequate, it is a subtle behavior change that I don't think the current specification covers.

I get the point now.

> As mentioned, I don't think this change is critical unless you have a stronger case.

I'll withdraw this PR.